### PR TITLE
webconsole refactoring with react 18

### PR DESCRIPTION
**Pull request by kishiguro**
webconsole refactoring with react 18 and typescript and material ui.

### CODE REVIEW

**Comment by brianchennn** [comment]
I can't login. Can you fix it?

image

File: frontend/WebUI-diff/WebUI.diff [hunk 1]
```diff
@@ -0,0 +1,234 @@
+diff --color -rc webconsole-orig/backend/WebUI/api_webui.go webconsole/backend/WebUI/api_webui.go
```
Comment (ianchen0119) [inline frontend/WebUI-diff/WebUI.diff:1]: Hi @kishiguro 
Do we need this patch file after PR merged?

**Comment by brianchennn** [comment]
@kishiguro
I think the UI is beautiful, but there's an error when test with UERANSIM
image
UE's setting:
image

File: frontend/WebUI-diff/WebUI.diff (hunk 1 above)
Comment (kishiguro) [reply frontend/WebUI-diff/WebUI.diff:1]: @ianchen0119 No, no need of merging the patch.  it is included in the PR.

**Comment by kishiguro** [comment]
@brianchennn are you able to login with User: free5gc  Pass: admin ?

**Comment by brianchennn** [comment]
There are some issues:
- How can I create 100 UEs simultaneously ?
- When I add Flow Rules, error happened.
- It isn't able to display the UEs information created by old webconsole.

**Comment by kishiguro** [comment]
@brianchennn
> Yes, I can login now.

nice!

> I think the UI is beautiful, but there's an error when test with UERANSIM.

Let me double check with the same configuration

**Comment by brianchennn** [comment]
@kishiguro 
Panic when add Flow Rules:
```
2023-08-01T11:34:35.547912779Z [INFO][SMF][Consumer] SendNFDiscoveryServingAMF ok
2023-08-01T11:34:35.548042234Z [INFO][SMF][CTX] Allocated UE IP address: 10.60.0.6
2023-08-01T11:34:35.548063530Z [INFO][SMF][CTX] Selected UPF: UPF
2023-08-01T11:34:35.548080809Z [INFO][SMF][PduSess][pdu_session_id:1][supi:imsi-208930000000997] Allocated PDUAdress[10.60.0.6]
2023-08-01T11:34:35.548279120Z [INFO][NRF][DISC] Handle NFDiscoveryRequest
2023-08-01T11:34:35.550266626Z [INFO][NRF][GIN] | 200 |       127.0.0.1 | GET     | /nnrf-disc/v1/nf-instances?preferred-locality=area1&requester-nf-type=SMF&target-nf-type=PCF |
2023-08-01T11:34:35.551690950Z [INFO][PCF][SMpolicy] Handle CreateSmPolicy
2023-08-01T11:34:35.555088331Z [INFO][UDR][DataRepo] Handle PolicyDataUesUeIdSmDataGet
2023-08-01T11:34:35.556130071Z [INFO][UDR][GIN] | 200 |       127.0.0.1 | GET     | /nudr-dr/v1/policy-data/ues/imsi-208930000000997/sm-data?dnn=internet&snssai=%7B%22sst%22%3A1%2C%22sd%22%3A%22010203%22%7D |
2023-08-01T11:34:35.557666677Z [ERRO][PCF][GIN] panic: interface conversion: interface {} is nil, not float64
goroutine 653 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/free5gc/util/logger.ginRecover.func1.1()
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:284 +0x1a5
panic({0xbf8fc0, 0xc0006177a0})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/free5gc/pcf/internal/sbi/producer.createSMPolicyProcedure({0x0, 0x0, {0x0, 0x0}, {0xc0005d20c0, 0x14}, {0x0, 0x0, 0x0}, 0x1, ...})
        /home/brian/free5gc-github/NFs/pcf/internal/sbi/producer/smpolicy.go:226 +0x2c45
github.com/free5gc/pcf/internal/sbi/producer.HandleCreateSmPolicyRequest(0xc000371b60)
        /home/brian/free5gc-github/NFs/pcf/internal/sbi/producer/smpolicy.go:37 +0x145
github.com/free5gc/pcf/internal/sbi/smpolicy.HTTPSmPoliciesPost(0xc000362700)
        /home/brian/free5gc-github/NFs/pcf/internal/sbi/smpolicy/api_default.go:56 +0x29a
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/free5gc/util/logger.ginRecover.func1(0xc000362700)
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:316 +0x7b
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/free5gc/util/logger.ginToLogrus.func1(0xc000362700)
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:251 +0x82
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc000242340, 0xc000362700)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/gin.go:620 +0x671
github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc000242340, {0xe8e318, 0xc000534028}, 0xc000362600)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/gin.go:576 +0x1dd
golang.org/x/net/http2.(*serverConn).runHandler(0x13ea320, 0x0, 0x0, 0xc0003faf00)
        /home/brian/go/pkg/mod/golang.org/x/net@v0.7.0/http2/server.go:2311 +0x8c
created by golang.org/x/net/http2.(*serverConn).processHeaders
        /home/brian/go/pkg/mod/golang.org/x/net@v0.7.0/http2/server.go:2025 +0x5f4
2023-08-01T11:34:35.557825062Z [INFO][PCF][GIN] | 500 |       127.0.0.1 | POST    | /npcf-smpolicycontrol/v1/sm-policies |
2023-08-01T11:34:35.558279343Z [ERRO][SMF][GIN] panic: interface conversion: interface {} is nil, not models.ProblemDetails
goroutine 338 [running]:
runtime/debug.Stack()
        /usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/free5gc/util/logger.ginRecover.func1.1()
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:284 +0x1a5
panic({0xaad1a0, 0xc0000e3380})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/free5gc/smf/internal/sbi/producer.HandlePDUSessionSMContextCreate({0xc00059c480, {0xc0006fec00, 0x12, 0x3e8}})
        /home/brian/free5gc-github/NFs/smf/internal/sbi/producer/pdu_session.go:169 +0x16fd
github.com/free5gc/smf/internal/sbi/pdusession.HTTPPostSmContexts(0xc0006d0700)
        /home/brian/free5gc-github/NFs/smf/internal/sbi/pdusession/api_sm_contexts_collection.go:54 +0x265
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/free5gc/util/logger.ginRecover.func1(0xc0006d0700)
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:316 +0x7b
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/free5gc/util/logger.ginToLogrus.func1(0xc0006d0700)
        /home/brian/go/pkg/mod/github.com/free5gc/util@v1.0.5-0.20230511064842-2e120956883b/logger/logger.go:251 +0x82
github.com/gin-gonic/gin.(*Context).Next(...)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/context.go:174
github.com/gin-gonic/gin.(*Engine).handleHTTPRequest(0xc0000836c0, 0xc0006d0700)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/gin.go:620 +0x671
github.com/gin-gonic/gin.(*Engine).ServeHTTP(0xc0000836c0, {0xc99d28, 0xc00052c1b0}, 0xc0006d0600)
        /home/brian/go/pkg/mod/github.com/gin-gonic/gin@v1.9.0/gin.go:576 +0x1dd
golang.org/x/net/http2.(*serverConn).runHandler(0xc000524930, 0xc0005757d0, 0x8a24a6, 0xc0004d9080)
        /home/brian/go/pkg/mod/golang.org/x/net@v0.7.0/http2/server.go:2311 +0x8c
created by golang.org/x/net/http2.(*serverConn).processHeaders
        /home/brian/go/pkg/mod/golang.org/x/net@v0.7.0/http2/server.go:2025 +0x5f4
2023-08-01T11:34:35.558543765Z [INFO][SMF][GIN] | 500 |       127.0.0.1 | POST    | /nsmf-pdusession/v1/sm-contexts |
2023-08-01T11:34:35.558940606Z [ERRO][AMF][Gmm][amf_ue_ngap_id:RU:8,AU:8(3GPP)][supi:SUPI:imsi-208930000000997] CreateSmContextRequest Error: undefined response type
```

**Comment by brianchennn** [comment]
@kishiguro 
After editing Flow Rules, these fields still empty except IP Filter:
image

**Comment by brianchennn** [comment]
@kishiguro
I want to let you know that another contributor is also modifying the webconsole in [PR #43](https://github.com/free5gc/webconsole/pull/43). Free5gc team will choose better one to merge.

**Comment by kishiguro** [comment]
@brianchennn Seems there is a bug of setting flow rules.  Let me fix it.

**Comment by kishiguro** [comment]
@brianchennn FlowRule (and QosFlows) handling has been fixed.  Would you mind to try again?

File: frontend/WebUI-diff/WebUI.diff (hunk 1 above)
Comment (kishiguro) [reply frontend/WebUI-diff/WebUI.diff]: @ianchen0119 diff has been removed to avoid confusion.

**Comment by brianchennn** [comment]
@kishiguro 
Thanks for fixing the bug. Now I can't add second DNN:
https://youtu.be/Rj9iRkLBZ6A

**Comment by kishiguro** [comment]
> @kishiguro Thanks for fixing the bug. Now I can't add second DNN: https://youtu.be/Rj9iRkLBZ6A

Thanks for pointing out.  Just fixed the issue.  Would you mind to check it out?

**Comment by brianchennn** [comment]
@kishiguro 
That's great! 
Next, can you help me to add this field for adding multiple UEs? Sometimes we may do that for testing. Thanks a lot !
image

**Comment by kishiguro** [comment]
@brianchennn 

> Next, can you help me to add this field for adding multiple UEs by increment SUPI? Sometimes we may do that for testing.

sure, wait a sec.

**Comment by kishiguro** [comment]
@brianchennn just added support for multiple UEs by increment SUPI.  would you mind to try?

**Comment by brianchennn** [comment]
@kishiguro  It works well ! 
Now, could you make the default S-NSSAI setting be 2 S-NSSAI Configuragtions as below 2 figure?
image
image

**Comment by kishiguro** [comment]
@brianchennn 

> Now, could you make the default S-NSSAI setting be 2 S-NSSAI Configuragtions as below?

Now default S-NSSAI configuartion will be set.  Would you mind to double check?

**Comment by brianchennn** [comment]
@kishiguro 
Could you add the **confirm password** field when creating/modifying user of tenant?
And the password field should be concealed in browser view.
I just added these features  in [PR #44](https://github.com/free5gc/webconsole/pull/44).
image

I also found that when creating a tenant, the hint of this field is **tenant ID**, but it should be **tenant name**.
image

**Comment by kishiguro** [comment]
@brianchennn got it let me integrate the confirm password feature.

**Comment by kishiguro** [comment]
@brianchennn I have integrated Password Confirmation feature.  "Tenant ID" is also fixed as "Tenant Name".

**Comment by brianchennn** [comment]
@tim-ywliu 
I think we can merge this PR.

**Review by tim-ywliu — APPROVED** [review]

**Pull request merged**